### PR TITLE
[usbdev/dif] Add missing DIFs

### PR DESCRIPTION
**Pull request by a-will**
- Clean up deprecated functions in the usbdev DIFs.
- Add missing DIFs for usbdev CSRs.
- Add unit tests for half the DIFs (more to come in a later PR).

### CODE REVIEW

**Comment by a-will** [comment]
Cool... `git clang-format --style=file` is not providing the same format on my workstation anymore.

File: sw/device/lib/dif/dif_usbdev.h [hunk 1]
```diff
@@ -199,7 +199,18 @@ dif_result_t dif_usbdev_endpoint_setup_enable(const dif_usbdev_t *usbdev,
...
+ * endpoint, disabling reception of OUT packets will cause them to be ignored.
+ *
+ * If the endpoint is neither a control nor isochronous endpoint, then its
+ * behavior depends on whether it is configured to respond with STALL. If the
+ * STALL response is not active, then disabling reception will cause usbdev to
+ * NAK the packet. Otherwise, the STALL response takes priority, regardless of
+ * the setting here.
  *
  * @param usbdev A USB device.
  * @param endpoint An OUT endpoint number.
```
Comment (timothytrippel) [inline sw/device/lib/dif/dif_usbdev.h:216]: nit:
```suggestion
 * @param[out] endpoint An OUT endpoint number.
```

Comment (timothytrippel) [reply sw/device/lib/dif/dif_usbdev.h:216]: (applies throughout)

**Review by timothytrippel — APPROVED** [review]

File: sw/device/lib/dif/dif_usbdev.h (hunk 1 above)
Comment (a-will) [reply sw/device/lib/dif/dif_usbdev.h:216]: Uh, that's not correct. It is not an output parameter. The "OUT" here refers to USB transaction directions.

Comment (timothytrippel) [reply sw/device/lib/dif/dif_usbdev.h:216]: oops sorry you are right, the way it looked on my screen i thought this comment applied to  `dif_result_t dif_usbdev_address_get()`... the code blocked was collapsed. My apologies!

**Review by timothytrippel — APPROVED** [review]

File: sw/device/lib/dif/dif_usbdev.h (hunk 1 above)
Comment (a-will) [reply sw/device/lib/dif/dif_usbdev.h:216]: Hm... But it does look like there are some issues for the older code.

Comment (timothytrippel) [reply sw/device/lib/dif/dif_usbdev.h:216]: SG. yeah I saw that but didn't think it was fair to call out as that was not your change. But if you feel inclined to fixed, by all means hehe. Thanks @a-will !

**Pull request merged**